plugins/x86/x86_prefix.mli

```diff
@@ -0,0 +1,22 @@
+open Core_kernel.Std
+open Bap.Std
+       
+type t = [
+  | `_F0 (** LOCK                *)
+  | `_F2 (** REPNE               *)
+  | `_F3 (** REP/REPE            *)
+  | `_2E (** CS segment override *)
+  | `_36 (** SS segment override *)
+  | `_3E (** DS segment override *)
+  | `_26 (** ES segment override *)
+  | `_64 (** FS segment override *)
+  | `_65 (** GS segment override *)
+  | `_66 (** operand override    *)
+  | `_67 (** address override    *)
+] [@@deriving bin_io, sexp, compare, enumerate]
+
+(** [get mem] - returns a list of prefixes, if any *)
+val get : mem -> t list Or_error.t
```

Comment:
I don't think that there is any reason to return an error here. Since prefixes are totally optional, the absence of a prefix should never be considered as an error. You basically abusing here the error type, and treating it as an option.